nvflare/fuel/f3/stats_pool.py

```diff
@@ -501,34 +501,66 @@ class CsvRecordHandler(RecordWriter):
     def __init__(self, file_name):
         self.file = open(file_name, "w")
         self.writer = csv.writer(self.file)
+        self.lock = threading.Lock()
 
     def write(self, pool_name: str, category: str, value: float, report_time: float):
+        if not pool_name.isascii():
+            raise ValueError(f"pool_name {pool_name} contains non-ascii chars")
+        if not category.isascii():
+            raise ValueError(f"category {category} contains non-ascii chars")
+
         row = [pool_name, category, report_time, value]
-        self.writer.writerow(row)
+        with self.lock:
+            self.writer.writerow(row)
```

Comment:
What will happen if reading the csv file and write at the same time, since lock only applies to the write()? 
Probably it will be okay. Just wondering, is it worth to explore using the logger to write to the csv file? Logger should handle the thread reading / writing internally already.

Reply:
We don't have this use case. The writing happens during job execution. Reading happens after the job is done for post analysis. We can't access the csv file before the job is done since the file isn't even closed, so the text could be incomplete.

Reply:
I don't think we should use logger to write csv files. Even though CSV stands for "comma separated value", the actual separator doesn't have to be commas. Python's csv library can handle all cases, including escaping when value itself contains separators.